### PR TITLE
SegmentEngine 1.0.0

### PR DESCRIPTION
**Pull request by nnlamb25**
@CMBleakley for review

### CODE REVIEW

**Comment by nnlamb25** [comment]
For [#3197](https://github.com/Factual/engine/issues/3197)

File: README.md [hunk 1]
```diff
@@ -34,119 +24,87 @@ repositories {
...
+    /*
+    Max number of "engine_span_attached_place" events that should be sent per
+    "engine_span_occurred". Default is set to 20.
+    */
+    int maxAttachedPlaceEventsPerSpan = 10;
+
+    /* Start tracking spans */
+    SegmentEngineIntegration.trackUserJourneySpans(getContext().getApplicationContext(),
+        maxAttachedPlaceEventsPerSpan);
+}
```
Comment (artstwolf) [inline README.md]: The indentation is off here, right?

File: example/src/androidTest/java/com/factual/engine/segment/SegmentEngineIntegrationTest.java [hunk 2]
```diff
@@ -0,0 +1,329 @@
...
+      fail("Could not create placeObject because of exception: " + exception.getMessage());
+    }
+
+    FactualPlaceVisit placeVisit = new FactualPlaceVisit(location,
+        places,
+        null,
+        false,
+        false);
+
+    return new UserJourneySpan(
```
Comment (artstwolf) [inline example/src/androidTest/java/com/factual/engine/segment/SegmentEngineIntegrationTest.java]: You can use the `UserJourneySpan.Builder`.

File: library/src/main/java/com/factual/engine/segment/SegmentEngineIntegration.java [hunk 3]
```diff
@@ -0,0 +1,216 @@
...
+import android.util.Log;
+import com.factual.engine.api.CircumstanceResponse;
+import com.factual.engine.api.FactualPlace;
+import com.segment.analytics.Analytics;
+import com.segment.analytics.Properties;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.UUID;
+
+public class SegmentEngineIntegration {
```
Comment (artstwolf) [inline library/src/main/java/com/factual/engine/segment/SegmentEngineIntegration.java:19]: You should add a class level javadoc comment.

**Review by artstwolf — COMMENTED** [review]
I left a few comments inline.

**Pull request merged**